src/main/tut/2.4-representable-functors.md

```diff
@@ -0,0 +1,208 @@
+```Haskell
+C(a, x) -> C(a, y)
+```
+```scala
+`C(A, X)` => `C(A, Y)`
+```
+................
+```Haskell
+f ∘ h :: a -> y
+```
+```scala
+val `f ∘ h`: A => Y
+```
+................
+```Haskell
+type Reader a x = a -> x
+```
+```tut:silent
+case class Reader[A, X](run: A => X)
+```
+................
+```Haskell
+instance Functor (Reader a) where
+    fmap f h = f . h
+```
+```scala
+implicit def readerFunctor[A] =
+  new Functor[({type T[X] = Reader[A, X]})#T] {
+    def fmap[X, B](f: X => B)(h: Reader[A, X])
+        : Reader[A, B] =
+      Reader(x => f(h.run(x)))
+  }
+
+/* type lambda with Kind Projector plugin:
+   new Functor[Reader[A, ?]] {...} */
+```
+................
+```Haskell
+type Op a x = x -> a
+```
+```tut:silent
+case class Op[A, X](f: X => A)
+```
+................
+```Haskell
+instance Contravariant (Op a) where
+    contramap f h = h . f
+```
+```scala
+implicit def opContravariant[A] =
+  new Contravariant[({type T[X] = Op[A, X]})#T] {
+    def contramap[X, B](f: B => X)
+        : Op[A, X] => Op[A, B] = {
+      case Op(h) => Op(h compose f)
+    }
+  }
+
+/* type lambda with Kind Projector plugin:
+   new Contravariant[Op[A, ?]] {...} */
+```
+................
+```Haskell
+instance Profunctor (->) where
+  dimap ab cd bc = cd . bc . ab
+  lmap = flip (.)
+  rmap = (.)
+```
+```scala
+implicit def arrowProfunctor =
+  new Profunctor[Function1] {
+    def dimap[A, B, C, D]
+        (ab: A => B)(cd: C => D)(bc: B => C)
+        : A => D =
+      cd compose bc compose ab
+
+    def lmap[A, B, C]
+        (f: C => A)(ab: A => B): C => B =
+      f andThen ab
+
+    def rmap[A, B, C]
+        (f: B => C)(ab: A => B): A => C =
+      f compose ab
+  }
+```
+................
+```Haskell
+C(-, =) :: Cop × C -> Set
+```
+```scala
+val `C(-, =)`: (Cop `×` C) => Set
+```
+................
+```Haskell
+αx :: C(a, x) -> F x
+```
+```scala
+val αx: `C(A, X)` => F[X]
```

Comment:
`A` and `X` are type variables in the haskell code, in yours `C(A, X)` is a concrete type named `C(A, X)` (there the `X` has no connection with the latter `X`), and `X` is a concrete type called `X`